### PR TITLE
Add TABLE_VALUED_FUNCTION and returnTableType to bigquery_routine

### PR DESCRIPTION
**Pull request by Tei1988**
Add TABLE_VALUED_FUNCTION and returnTableType to bigquery#routine.

This PR is contributes to https://github.com/hashicorp/terraform-provider-google/issues/10120 .

I just tested to create a table valued function using local built provider.
But I can not run acceptance tests.

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
bigquery: added `TABLE_VALUED_FUNCTION` to routine_type on resource `bigquery_routine`
bigquery: added field `return_table_type` to resource `bigquery_routine`
```

### CODE REVIEW

**Comment by rileykarson** [comment]
Running tests! The first pass will have the new test fail, but the second should succeed.

/gcbrun

**Comment by Tei1988** [comment]
@rileykarson
Thank you for running tests.

I'm not sure what should I do after this.
Should I fix something ?

**Comment by rileykarson** [comment]
Hmm, seems they didn't get posted. Let me try again.

/gcbrun

**Comment by Tei1988** [comment]
@rileykarson 
Thank you for rerunning tests.

Does this fail mean the first pass you said before?
I tried to check the above url, but It seems that I can't access it.

**Comment by rileykarson** [comment]
Yep- I can see the logs, and `TestAccBigQueryRoutine_bigQueryRoutineTvfExample` passed on the second run. Everything else in that list are false positives, effectively.

Note: TFV failures are unrelated.

**Review by rileykarson — APPROVED** [review]
